### PR TITLE
Mmi 4364 only get rewards for open lp positions

### PR DESCRIPTION
**Pull request by jpsains**

### CODE REVIEW

File: src/adapters/convex/products/staking/convexStakingAdapter.ts [hunk 1]
```diff
@@ -55,6 +61,16 @@ export class ConvexStakingAdapter
...
 
+  @AddClaimableRewards({ rewardAdapterIds: ['extra-reward', 'rewards'] })
+  getPositions(input: GetPositionsInput): Promise<ProtocolPosition[]> {
+    return super.getPositions(input)
+  }
+
+  @AddClaimedRewards({ rewardAdapterIds: ['extra-reward', 'rewards'] })
+  getWithdrawals(input: GetEventsInput): Promise<MovementsByBlock[]> {
+    return super.getWithdrawals(input)
+  }
```
Comment (jpsains) [inline src/adapters/convex/products/staking/convexStakingAdapter.ts:72]: Add claim**ed** rewards to withdrawals

This ensures they are included in our pnl calculations

File: src/adapters/convex/tests/snapshots/ethereum.positions.2.json [hunk 2]
```diff
@@ -32,48 +32,6 @@
           "balanceRaw": "28941976633661456319n",
```
Comment (jpsains) [inline src/adapters/convex/tests/snapshots/ethereum.positions.2.json:32]: Looked several times are imo just the order has changed

File: src/adapters/curve/products/staking/curveStakingAdapter.ts [hunk 3]
```diff
@@ -216,6 +225,7 @@ export class CurveStakingAdapter
   }
 
   @ResolveUnderlyingMovements
+  @AddClaimedRewards({ rewardAdapterIds: ['reward'] })
```
Comment (jpsains) [inline src/adapters/curve/products/staking/curveStakingAdapter.ts:228]: we resolve the underlying first then add claimed rewards

File: src/adapters/curve/tests/snapshots/ethereum.profits.json [hunk 4]
```diff
@@ -44,20 +44,6 @@
...
-      "siteUrl": "https://curve.fi/",
-      "iconUrl": "https://raw.githubusercontent.com/MetaMask/contract-metadata/master/images/crv.svg",
-      "positionType": "reward",
-      "chainId": 1,
-      "productId": "reward",
-      "success": true,
-      "tokens": [],
-      "fromBlock": 16621005,
-      "toBlock": 16628151
     }
```
Comment (jpsains) [inline src/adapters/curve/tests/snapshots/ethereum.profits.json:47]: Perfect no longer returning rewards adapters

File: src/core/adaptersController.ts [hunk 5]
```diff
@@ -37,11 +40,6 @@ export class AdaptersController {
           ([chainIdKey, adapterClasses]) => {
             const chainId = +chainIdKey as Chain
 
-            // always use mainnet for price adapters
-            // const provider =
-            //   protocolId == Protocol.Prices
-            //     ? providers[Chain.Ethereum]!
-            //     : providers[chainId]!
```
Comment (jpsains) [inline src/core/adaptersController.ts:44]: Slipped in by accident in prices mr

File: src/core/adaptersController.ts [hunk 6]
```diff
@@ -95,54 +93,81 @@ export class AdaptersController {
...
+          switch (positionType) {
+            case PositionType.FiatPrices:
+              this.processFiatPrices(protocolTokens, chainAdaptersMap, adapter)
+              break
+            case PositionType.Reward:
+              // Handle Reward case here if needed
+              break
+            default:
+              this.processDefaultCase(protocolTokens, chainAdaptersMap, adapter)
+              break
```
Comment (jpsains) [inline src/core/adaptersController.ts:128]: Not related to this feature, this is an additional refactor to improve readability

File: src/core/adaptersController.ts [hunk 7]
```diff
@@ -95,54 +93,81 @@ export class AdaptersController {
...
+      }
+    }
+  }
+
+  private processDefaultCase(
+    protocolTokens: Erc20Metadata[],
+    chainAdaptersMap: Map<string, IProtocolAdapter>,
+    adapter: IProtocolAdapter,
+  ) {
+    for (const protocolToken of protocolTokens) {
```
Comment (jpsains) [inline src/core/adaptersController.ts:156]: Note change is not related to this feature. Refactor to improve readability

File: src/adapters/curve/products/staking/curveStakingAdapter.ts (hunk 3 above)
Comment (bergarces) [reply src/adapters/curve/products/staking/curveStakingAdapter.ts:228]: I don't think there's any way to enforce the order via typescript, so we'll have to keep an eye in future cases.

File: src/adapters/curve/tests/snapshots/ethereum.profits.profits-2.json [hunk 8]
```diff
@@ -32,43 +32,13 @@
           "symbol": "3Crv",
           "decimals": 18,
           "type": "protocol",
-          "profit": 0,
-          "performance": 0,
+          "profit": 3.15529238292811e-8,
+          "performance": 4.90916972549114e-9,
```
Comment (bergarces) [inline src/adapters/curve/tests/snapshots/ethereum.profits.profits-2.json:36]: I'm going to assume this is somehow intended, but just want to double check.

Before the reward adapter was returning independently (and without the `fromBlock` and `toBlock`) and now it is part of the profits response.

File: src/core/adaptersController.ts [hunk 9]
```diff
@@ -13,6 +13,9 @@ export class AdaptersController {
   private protocolTokens:
     | Promise<Map<Chain, Map<string, IProtocolAdapter>>>
     | undefined
+  private rewardTokens:
```
Comment (bergarces) [inline src/core/adaptersController.ts:16]: I thought this wasn't needed with the decorator.

File: src/core/decorators/addClaimableRewards.ts [hunk 10]
```diff
@@ -0,0 +1,53 @@
...
+            protocolTokens.map(async (protocolToken) => {
+              const [reward] = await rewardAdapter.getPositions({
+                ...input,
+                protocolTokenAddresses: [protocolToken.address],
+              })
+
+              if (reward) {
+                if (reward.tokens) {
+                  protocolToken.tokens = [
+                    ...(protocolToken?.tokens || []),
```
Comment (bergarces) [inline src/core/decorators/addClaimableRewards.ts:38]:
```suggestion
                    ...(protocolToken.tokens ?? []),
```
- `protocolToken` is 100% defined, so no need for the `?`.
- nit, and in this case it doesn't matter, but `??` is preferable over `||`, as `??` acts when the value is `undefined`, but `||` acts when the value is falsy.

File: src/core/decorators/addClaimedRewards.ts
Comment (bergarces) [inline src/core/decorators/addClaimedRewards.ts:1]: image

File: src/adapters/convex/tests/snapshots/ethereum.positions.2.json (hunk 2 above)
Comment (jpsains) [reply src/adapters/convex/tests/snapshots/ethereum.positions.2.json:32]: nice idea

File: src/adapters/curve/products/staking/curveStakingAdapter.ts (hunk 3 above)
Comment (jpsains) [reply src/adapters/curve/products/staking/curveStakingAdapter.ts:228]: good point i wonder if the order matters, might actually get the same results either way 🤔

File: src/core/adaptersController.ts (hunk 9 above)
Comment (jpsains) [reply src/core/adaptersController.ts:16]: ah let me remove, good catch
